src/games/strategy/triplea/attachments/TerritoryAttachment.java

```diff
@@ -138,15 +138,7 @@ public static Territory getFirstOwnedCapitalOrFirstUnownedCapital(final PlayerID
    * Convenience method. Can return null.
    */
   public static TerritoryAttachment get(final Territory t) {
-    return get(t, false);
-  }
-
-  public static TerritoryAttachment get(final Territory t, final boolean allowNull) {
-    final TerritoryAttachment rVal = (TerritoryAttachment) t.getAttachment(Constants.TERRITORY_ATTACHMENT_NAME);
-    if (!allowNull && rVal == null && !t.isWater()) {
-      throw new IllegalStateException("No territory attachment for:" + t.getName());
-    }
-    return rVal;
+    return (TerritoryAttachment) t.getAttachment(Constants.TERRITORY_ATTACHMENT_NAME);
```

Comment:
Right my concern is that the method above calls it with 'false' which is probably the method that is called publicly more often. That would make me think that we are relying on the exception being thrown in some cases. While you are correct that you'd most likely end up with NPE instead that would most likely be less descriptive of what the problem is.